scripts/cbmc_json_parser.py

```diff
@@ -255,6 +258,129 @@ def filter_properties(properties, message):
+        CProverCheck("pointer arithmetic: pointer outside dynamic object bounds"),
+        CProverCheck("pointer arithmetic: pointer outside object bounds"),
+        CProverCheck("pointer arithmetic: invalid integer address")],
+    "pointer_dereference": [
+        CProverCheck("dereferenced function pointer must be", "dereference failure: invalid function pointer"),
+        CProverCheck("dereference failure: pointer NULL"),
+        CProverCheck("dereference failure: pointer invalid"),
+        CProverCheck("dereference failure: deallocated dynamic object"),
+        CProverCheck("dereference failure: dead object"),
+        CProverCheck("dereference failure: pointer outside dynamic object bounds"),
+        CProverCheck("dereference failure: pointer outside object bounds"),
+        CProverCheck("dereference failure: invalid integer address")],
+    "pointer_primitives": [
+        # These are very hard to understand without more context.
+        CProverCheck("pointer invalid"),
+        CProverCheck("deallocated dynamic object", "pointer to deallocated dynamic object"),
+        CProverCheck("dead object", "pointer to dead object"),
+        CProverCheck("pointer outside dynamic object bounds"),
+        CProverCheck("pointer outside object bounds"),
+        CProverCheck("invalid integer address")
+    ],
+    "array_bounds": [
+        CProverCheck("lower bound", "access out of bounds"),
+        # This one is redundant: CProverCheck("dynamic object upper bound", "access out of bounds"),
+        CProverCheck("upper bound", "access out of bounds"), ],
+    "bit_count": [
+        CProverCheck("count trailing zeros is undefined for value zero"),
+        CProverCheck("count leading zeros is undefined for value zero")],
+    "memory-leak": [
+        CProverCheck("dynamically allocated memory never freed")],
+    # These pre-conditions should not print temporary variables since they are embedded in the libc implementation.
+    # They are added via __CPROVER_precondition.
+    # "precondition_instance": [],
+}
+
+def get_friendly_description(prop):
+    """This function will return a user friendly property description.
+
+       For CBMC checks, it will ensure that the failure does not include any
+       temporary variable.
+    """
+    original = prop["description"]
+    # Id is structured as [<function>.]<property_class>.<counter>
+    prop_class = prop["property"].rsplit(".", 3)
+    # Do nothing if prop_class is diff than cbmc's convention
+    class_id = prop_class[-2] if len(prop_class) > 1 else None
+    if class_id in CBMC_DESCRIPTIONS:
+        prop_type = [check.replace(original) for check in CBMC_DESCRIPTIONS[class_id] if check.matches(original)]
+        if len(prop_type) != 1:
+            if "KANI_FAIL_ON_UNEXPECTED_DESCRIPTION" in os.environ:
+                print("Unexpected description: {}".format(original))
+                print(f"  - class_id: {class_id}")
+                print(f"  - matches: {[prop_type]}")
```

Comment:
The array brackets in `{[prop_type]}` is unnecessary. It would be printed as - `[["apple", "oranges"]]` for example. You can replace it with just `{prop_type}`.

Reply:
My bad. This variable used to be an iterator.